src/PolygonDraw/Map.tsx

```diff
@@ -183,19 +189,67 @@ export class BaseMap extends React.Component<Props, State> {
         }
     };
 
-    handleMouseMoveOnMap = (event: LeafletMouseEvent) => {
+    handleMouseDownOnMap = (event: LeafletMouseEvent) => {
         const coordinate = createCoordinateFromLeafletLatLng(event.latlng);
-        const newPointPosition =
-            this.state.isPenToolActive &&
-            !this.props.isPolygonClosed &&
-            isCoordinateInPolygon(coordinate, this.props.boundaryPolygonCoordinates)
-                ? coordinate
-                : null;
 
-        this.setState({ newPointPosition });
+        if (this.state.isShiftPressed) {
+            this.setState({
+                isRectangleSelectionDragActive: true,
+                rectangleSelectionStartPosition: coordinate,
+                rectangleSelectionEndPosition: coordinate
+            });
+        }
+    };
+
+    handleMouseUpOnMap = () => {
+        if (this.state.isRectangleSelectionDragActive) {
```

Comment:
I'm not certain about this, but the condition could keep the rectangle visible if the user releases the shift key before releasing the mouse click.

Reply:
Yeah I did that intentionally as an affordance for the user, so they don't have to hold down the shift key. I also didn't see any useful behaviour you could get out of stopping the drag if the shift key is released before the mouse is.

Reply:
Sounds good. LGTM